test/es-module/test-esm-type-field-errors-2.js

```diff
@@ -0,0 +1,17 @@
+// Flags: --no-experimental-require-module
+// This used to test require(esm) throws ERR_REQUIRE_ESM, which is no longer the case
+// since require(esm) has been implemented. It's now repurposed to test that the
+// previous behavior can be kept when  --no-experimental-require-module is passed.
```

Comment:
```suggestion
```

This is a new file, is this notice really needed?

---

```suggestion
// Previously, this tested that require(esm) throws ERR_REQUIRE_ESM, which is no longer applicable 
// since require(esm) is now supported. The test has been repurposed to ensure that the old behavior 
// is preserved when the --no-experimental-require-module flag is used.
```

If so, suggestion on wording ☝️